src/ray/raylet/main.cc

```diff
@@ -21,7 +21,7 @@ int main(int argc, char *argv[]) {
                                          ray::RayLogLevel::INFO,
                                          /*log_dir=*/"");
   ray::RayLog::InstallFailureSignalHandler();
-  RAY_CHECK(argc >= 14 && argc <= 18);
+  RAY_CHECK(argc >= 14 && argc <= 19);
```

Comment:
Can we remove the upper bound on this RAY_CHECK? It causes crashes when the python files get out of sync.

Reply:
I'd like to reserve the upper bound on this CHECK.

Reply:
I agree with @jovany-wang. If the files are out of sync, then any number of things could go wrong and we probably want to make the user aware of this as quickly as possible. Failing more quickly also helps catch bugs during development.

Reply:
As a developer of the frontend libraries (and I'm sure others feel this pain too), it's very inconvenient when the backend breaks compatibility with the underlying C++ binaries. In the vast majority of these cases, it does not have to.

I'm not sure what the right solution is, perhaps the core python package should be split in two or something?

Reply:
@ericl is the issue that after doing a `git pull` you would prefer not to recompile Ray because it takes too long? Installation has gotten a lot faster now that we no longer compile Arrow.

Also, I'm not sure what it would mean to split the package in two. Can you elaborate?